hack/man-page-checker

```diff
@@ -99,9 +99,16 @@ function compare_usage() {
     from_man=$(sed  -E -e 's/^\[\*options\*\][[:space:]]*//' <<<"$from_man")
     from_help=$(sed -E -e 's/^\[command options\][[:space:]]*//'      <<<"$from_help")
 
-    # Args in man page are '*foo*', in --help are 'FOO'. Convert all to
-    # UPCASE simply because it stands out better to the eye.
-    from_man=$(sed -E -e 's/_([a-z-]+)_/\U\1/g' <<<"$from_man")
+    # Constant strings in man page are '**foo**', in --help are 'foo'.
+    from_man=$(sed -E -e 's/\*\*([^*]+)\*\*/\1/g' <<<"$from_man")
+
+    # Args in man page are '*foo*', in --help are 'FOO'. Selectively
+    # converting to upper-case is too hard, so just convert everything
+    # to upper case below to compare case-insensitively.
+    from_man=$(sed -E -e 's/_([a-z-]+)_/\1/g' <<<"$from_man")
+
+    from_man=$(tr '[:lower:]' '[:upper:]' <<<"$from_man")
+    from_help=$(tr '[:lower:]' '[:upper:]' <<<"$from_help")
```

Comment:
Real quick note about this: I deliberately wanted `from_help` to remain as-is, because the convention is for `--help` args to be UPPER CASE. If they are lower case, I wanted to flag that. (But it's late in the day, and I may not be fully understanding your purpose here)

Reply:
This is almost certainly too broad - the `--src` / `--dest` literals from `skopeo sync` should not be modified.

I just wanted have _something_ so that the other commits could be at least partially verified, and I can’t think of any simple replacement for `\U`.

Reply:
Alternatively, it might be worth it to just run this on Linux only, and let macOS users deal with CI-only failures that don’t show up locally.

Reply:
The more I think about it, the better that sounds. I think it's a waste of our time to work around Mac bugs, especially when some of us don't even have access to Macs. And I can't imagine a scenario in which man-page validation would catch something on a Mac that wouldn't be caught on Linux.

Reply:
I have dropped the macOS workaround for now, leaving `make validate-local` failing on macOS, so that this doesn’t conflict with the work in #1350.